### PR TITLE
Test/17/test for text

### PR DESCRIPTION
**Pull request by NazarPolit**
dev

## Code reviewers

- [ ] @VolodymyrShapoval 
- [ ] @ArturOleksiuk 
- [ ] @Vektor19 
- [x] @shribakb1 
- [x] @LausDio 
- [x] @Ammoniy 

### Second Level Review

- [ ] @DrFaust555 

## Summary of issue

Lack of tests for MediatR/StreetCode/Text functionality.

## Summary of change

Added unit tests for MediatR/StreetCode/Text :

GetAllTextsHandlerTests
GetTextByIdHandlerTests
GetTextByStreetcodeIdHandlerTests
GetParsedTextAdminPreviewHandlerTests

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: Streetcode/Streetcode.XUnitTest/TextTests/GetAllTextsHandlerTests.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+			// Arrange
+			_mockRepoWrapper.Setup(r => r.TextRepository.GetAllAsync(null, null))
+				.ReturnsAsync((IEnumerable<Text>?)null);
+
+			// Act
+			var result = await _handler.Handle(new GetAllTextsQuery(), CancellationToken.None);
+
+			// Assert
+			result.IsFailed.Should().BeTrue();
+			result.Errors.First().Message.Should().Be("Cannot find any text");
```
Comment (Ammoniy) [inline Streetcode/Streetcode.XUnitTest/TextTests/GetAllTextsHandlerTests.cs]: Мабуть краще цю строку винести як константу окремо, так само як і наступну в Handle_ShouldReturnFail_WhenTextDoesNotExist

**Review by Ammoniy — CHANGES_REQUESTED** [review]
В цілому добре тести працюють. Але ну Дуууже багато ворнінгів, ледви не весь код підкреслений, здебільшого по формату, не бачу сенсу все описувати, IDE мала б підсвітити, Богдан на дещо наголосив. Поправити форматування та й мабуть буде добре

File: Streetcode/Streetcode.XUnitTest/TextTests/GetParsedTextAdminPreviewHandlerTests.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            // Act
+            var result = await this.handler.Handle(command, CancellationToken.None);
+
+            // Assert
+            result.IsSuccess.Should().BeTrue();
+            result.Value.Should().Be(ParsedText);
+        }
+
+        [Fact]
+        public async Task Handle_ShouldReturnFail_WhenParsingReturnsNull()
```
Comment (LausDio) [inline Streetcode/Streetcode.XUnitTest/TextTests/GetParsedTextAdminPreviewHandlerTests.cs]: You should add tests for:
- an empty string
- special characters(for example, if someone will place a piece of code instead of regular text)
- long texts

File: Streetcode/Streetcode.XUnitTest/TextTests/GetAllTextsHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        }
+
+        [Fact]
+        public async Task Handle_ShouldReturnOk_WhenTextsExist()
+        {
+            // Arrange
+            var textsList = new List<Text> { new Text { Id = 1 }, new Text { Id = 2 } };
+            var textsDtoList = new List<TextDTO> { new TextDTO { Id = 1 }, new TextDTO { Id = 2 } };
+
+            this.mockRepoWrapper.Setup(r => r.TextRepository.GetAllAsync(null, null))
```
Comment (LausDio) [inline Streetcode/Streetcode.XUnitTest/TextTests/GetAllTextsHandlerTests.cs]: You should add mock verification.

**Review by LausDio — CHANGES_REQUESTED** [review]
Great work! Tests are solid. But before approval, please add edge cases and mock verification calls. Also, add class-level XML documentation to:
- `GetTextByIdHandlerTests.cs`
- `GetParsedTextAdminPreviewHandlerTests.cs` 
- `GetTextByStreetcodeIdHandlerTests.cs`

**Review by LausDio — APPROVED** [review]
That's better, thanks.
